### PR TITLE
Setup bazel

### PR DESCRIPTION
**Pull request by supersonicbyte**

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Enable Bzlmod for every Bazel command
+common --enable_bzlmod
```
Comment (adincebic) [inline .bazelrc]: No lomnger needed.

File: .bazelrc [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+common --remote_cache=grpcs://remote.buildbuddy.io
```
Comment (adincebic) [inline .bazelrc:1]: Do you mind creating us a buildbuddy account? They provide their services for open source projects

File: BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,55 @@
+load(
+    "@rules_xcodeproj//xcodeproj:defs.bzl",
+    "top_level_target",
+    "xcodeproj",
+)
+load("@gazelle//:def.bzl", "gazelle", "gazelle_binary")
```
Comment (adincebic) [inline BUILD.bazel]: We don't need gazelle.

**Review by adincebic — APPROVED** [review]
We need to gitignore .xcodeproj since it contains absolute paths.
Also add a .bazelignore and make sure that .git directory is not in bazel output base.

File: .bazelrc (hunk 2 above)
Comment (supersonicbyte) [reply .bazelrc:1]: Already created, we can setup the stuff in a different PR.

Comment (adincebic) [reply .bazelrc:1]: > Already created, we can setup the stuff in a different PR.

Perfect!

**Review by adincebic — APPROVED** [review]

**Pull request merged**